feat: mobile ui svg icons

fbwoolf opened the pull request:
This PR relocates all svgs from the mobile app into the UI library. They are now exported from the UI library as React components with a default size of 24x24 and a 'small' variant of 16x16. All other SvgProps can be passed to change color, size, etc. The theme is used in the UI library to set the 'currentColor' as 'ink.action-primary-default', so the icons work with dark mode.

The native UI library build is now using tsup (aka 'sup'). 😆 Once this is merged, I can refactor the existing 'web' icons to use the svgs, but I wanted to separate the work into two PRs.

Preview of use...
<img width="394" alt="Screenshot 2024-08-15 at 4 21 42 PM" src="https://github.com/user-attachments/assets/e54954a5-cd4c-4118-b4c7-c48e1e521aa6">

![signal-2024-08-18-192320_002](https://github.com/user-attachments/assets/344d3abe-f57c-4ad4-b917-73042deaf2d2)

![signal-2024-08-18-192320_003](https://github.com/user-attachments/assets/985941e1-6366-43dc-bf74-92c23f727afc)

kyranjamie commented on packages/ui/app.json:
```diff
@@ -31,6 +31,17 @@
+      "expo-font",
+      {
+        "fonts": [
+          "../assets-native/fonts/SpaceMono-Regular.ttf",
+          "../assets-native/fonts/ABCDiatype-Regular.otf",
+          "../assets-native/fonts/ABCDiatype-Light.otf",
+          "../assets-native/fonts/ABCDiatype-Medium.otf",
+          "../assets-native/fonts/marche-super-pro.otf"
```
Can we rename to be all the same casing?

fbwoolf replied:
Let me check this.

fbwoolf replied:
The fonts aren't broken that I can see, do we need them in both places?

fbwoolf replied:
Nvm, think I am wrong here.

kyranjamie commented on packages/ui/package.json:
```diff
@@ -4,12 +4,12 @@
   "scripts": {
-    "assets:native": "mkdir -p dist-native/src && cp -r src/assets-native ./dist-native/src/",
-    "assets:web": "mkdir -p dist-web/src && cp -r src/assets-web ./dist-web/src",
+    "assets:native": "mkdir -p dist-native/src && cp -r src/assets-native ./dist-native/src && cp -r src/assets ./dist-native/src",
+    "assets:web": "mkdir -p dist-web/src && cp -r src/assets-web ./dist-web/src && cp -r src/assets ./dist-web/src",
```
Not a fan of having these copy commands in the package.json. tsup doesn't have a "beforeBuild" like it does `onSuccess` but [per comment here](https://github.com/egoist/tsup/issues/1094#issuecomment-1987318528), anything in the build script runs before the config.

So, could we move these directory/asset copying actions there, using a node script to do the same?

fbwoolf replied:
I meant to come back to this bc came across a suggestion to use this plugin to copy assets in the tsup config:
https://github.com/LinbuduLab/esbuild-plugins/tree/main/packages/esbuild-plugin-copy

kyranjamie reviewed (approved):
Fantastic work Fara, this is a really tidy set up. Definitely worth the time you spent investigating this to get to this result.

pete-watters reviewed (approved):
Great work @fbwoolf 👏

kyranjamie commented on packages/ui/src/icons/arrow-left-icon.native.tsx:
```diff
@@ -0,0 +1,17 @@
+import ArrowLeft from '../assets/icons/arrow-left.svg';
+import { Icon, IconProps } from './icon/icon.native';
+
+export function ArrowLeftIcon({ variant, ...props }: IconProps) {
+  if (variant === 'small')
+    return (
+      <Icon {...props}>
+        <ArrowLeftSmall />
+      </Icon>
+    );
+  return (
+    <Icon {...props}>
+      <ArrowLeft />
+    </Icon>
+  );
+}
```
@edgarkhanzadian there are separate SVG files to handle small pixel rendering differences at smaller sizes

fbwoolf replied:
@edgarkhanzadian the designers asked us not to scale svgs b/w 16x16 and 24x24, so we have two separate exports for the 'default' and 'small' variants. If another size is needed, we can then scale from the default.

fbwoolf commented:
@kyranjamie I'm getting a 'not-to-dev' dependency error here with tsup. Docs say it should be installed as a dev dependency, but we had it as a dependency. I moved it and added `tsup` to the `pathNot`, let me know if that isn't the correct way to handle that error. I also gitignored the generated `.tsup` folder.

kyranjamie commented:
Config files like tsup shouldn't be subject to rules of depcruiser, so good to ignore (even globally)
